### PR TITLE
Property delegation in M13

### PR DESCRIPTION
**Pull request by ilya-g**

### CODE REVIEW

File: docs/reference/delegated-properties.md [hunk 1]
```diff
@@ -81,15 +81,27 @@ For a **mutable** property (a *var*{:.keyword}), a delegate has to _additionally
...
 * receiver --- same as for `get()`,
 * metadata --- same as for `get()`,
-* new value --- must be of the same type as a property or its subtype.
+* new value --- must be of the same type as a property or its supertype.
  
+`get()` and/or `set()` functions may be provided either as member functions of the delegate class or extension functions. The latter is handy when you need to delegate property to an object, which doesn't originally provide these functions.
+
+## Custom delegates
+
+If you're willing to provide a custom delegate you can implement either `ReadOnlyProperty`, or `ReadWriteProperty` interface from the standard library, though again it's not a requirement to implement these interfaces, they are provided only for convenience. One situation, in which they can be handful, is when you implement a delegate with an [object expression](object-declarations.html#object-expressions):
```
Comment (abreslav) [inline docs/reference/delegated-properties.md]: I'm not sure we should promote these interfaces or even have them at all

Comment (abreslav) [inline docs/reference/delegated-properties.md]: And it's "handy" or 'useful", not "handful" :)

**Comment by yole** [comment]
Aside from Andrey's comments, this looks good to me.

**Comment by ilya-g** [comment]
Ok, I've removed that part for the time being

**Pull request merged**